tfsdk/schema.go

```diff
@@ -15,21 +15,49 @@ import (
 var (
 	// ErrPathInsideAtomicAttribute is used with AttributeAtPath is called
 	// on a path that doesn't have a schema associated with it, because
-	// it's an element or attribute of a complex type, not a nested
+	// it's an element, attribute, or block of a complex type, not a nested
 	// attribute.
-	ErrPathInsideAtomicAttribute = errors.New("path leads to element or attribute of a schema.Attribute that has no schema associated with it")
+	ErrPathInsideAtomicAttribute = errors.New("path leads to element, attribute, or block of a schema.Attribute that has no schema associated with it")
 )
 
 // Schema is used to define the shape of practitioner-provider information,
 // like resources, data sources, and providers. Think of it as a type
 // definition, but for Terraform.
 type Schema struct {
-	// Attributes are the fields inside the resource, provider, or data
+	// Attributes are value fields inside the resource, provider, or data
 	// source that the schema is defining. The map key should be the name
 	// of the attribute, and the body defines how it behaves. Names must
-	// only contain lowercase letters, numbers, and underscores.
+	// only contain lowercase letters, numbers, and underscores. Names must
+	// not collide with any Blocks names.
+	//
+	// In practitioner configurations, an equals sign (=) is required to set
+	// the value. See also:
+	//   https://www.terraform.io/docs/language/syntax/configuration.html
+	//
+	// Attributes are strongly preferred over Blocks.
 	Attributes map[string]Attribute
 
+	// Blocks are structural fields inside the resource, provider, or data
+	// source that the schema is defining. The map key should be the name
+	// of the block, and the body defines how it behaves. Names must
+	// only contain lowercase letters, numbers, and underscores. Names must
+	// not collide with any Attributes names.
+	//
+	// Blocks are by definition, structural, meaning they are implicitly
+	// required in values.
+	//
+	// In practitioner configurations, an equals sign (=) cannot be used to
+	// set the value. Blocks are instead repeated as necessary, or require
+	// the use of dynamic block expressions. See also:
+	//   https://www.terraform.io/docs/language/syntax/configuration.html
+	//   https://www.terraform.io/docs/language/expressions/dynamic-blocks.html
+	//
+	// Attributes are strongly preferred over Blocks. Blocks should only be
+	// used for configuration compatibility with previously existing schemas
+	// from an older Terraform Plugin SDK. Efforts should be made to convert
+	// Blocks to Attributes as a breaking change for practitioners.
+	Blocks map[string]Block
```

Comment:
Because we know we want to remove this eventually, do we want to put a `// Deprecated:` comment on it so that linters pick up usage of it and people can have automated guidance in moving away?

Reply:
Ah yes, thank you, will add that.